lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -244,14 +250,15 @@ def validate_definitions(resources)
 
     def discover_resources
       resources = []
+      crds = cluster_resource_discoverer.crds(@sync_mediator)
       @logger.info("Discovering templates:")
 
       Dir.foreach(@template_dir) do |filename|
         next unless filename.end_with?(".yml.erb", ".yml", ".yaml", ".yaml.erb")
 
         split_templates(filename) do |r_def|
           r = KubernetesResource.build(namespace: @namespace, context: @context, logger: @logger,
-                                       definition: r_def, statsd_tags: @namespace_tags)
+                                       definition: r_def, statsd_tags: @namespace_tags, crds: crds)
```

Comment:
We could capture the fact that a resource is a CRD here instead of inside `KubernetesResource.build`. This would avoid having to pass through the `crds` array, which for most resources is a useless argument. We would have to deal with the case where the deploy logic is hardcoded (e.g. `redis.rb`, `cloud_sql.rb`), but that won't be too much trouble.

This would necessitate adding a `KubernetesResource.build_cr` method or something to that effect